.github/workflows/onPullRequestCommit.yml

```diff
@@ -0,0 +1,91 @@
+name: Automatic Testing
+
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  push:
+    branches:
+      - master
+      - staging
+
+  # Replace pull_request with pull_request_target if you
+  # plan to use this action with forks, see the Limitations section
+  pull_request:
+    branches:
+      - master
+      - staging
+
+permissions:
+  checks: write
+  contents: write
+
+jobs:
```

Comment:
split the linter test out like we're doing on travis - it's better clearer for us developers if something failed logically or we just breaking some linter rule (also linter failures wouldn't block the system from running the functional tests)

You're also missing the `build-test` check:
![image](https://user-images.githubusercontent.com/11945337/201380675-5f5e4552-11be-4a71-992f-3f2587c0498b.png)

Reply:
changed the order of the events so that it'll always try and run the tests before the lint, the actions will stop if one of the previous steps fails so it'll install, test then lint.

Reply:
why are you so against splitting the linter as a separate job? 😆 

even if you  keep the linter with the backend/frontend test `yarn build:test` on the front end should still be a separate job - thanks to type script type checking that job takes like 15mins so we would want to run that in parallel

Reply:
ok